modules/kubernetes/fluxcd-v2/main.tf

```diff
@@ -0,0 +1,205 @@
+ * the bootstrap configuration.
+ *
+ * <p align="center"><img src="../../../assets/fluxcd-v2.jpg"></p>
+ */
+
+terraform {
+  required_version = "0.13.5"
+
+  required_providers {
+    flux = {
+      source  = "fluxcd/flux"
+      version = "0.0.4"
+    }
+    azuredevops = {
+      source  = "xenitab/azuredevops"
+      version = "0.2.1"
+    }
+    kubernetes = {
+      source  = "hashicorp/kubernetes"
+      version = "1.13.3"
+    }
+    kubectl = {
+      source  = "gavinbunney/kubectl"
+      version = "1.9.1"
+    }
+  }
+}
+
+locals {
+  repo_url = "http://git-cache-http-server/dev.azure.com/${var.azdo_org}/${var.azdo_proj}/_git/${var.azdo_repo}"
+}
+
+# FluxCD
+data "flux_install" "main" {
+  target_path = var.git_path
+}
+
+data "flux_sync" "main" {
+  url         = local.repo_url
+  target_path = var.git_path
+  branch      = var.branch
+  interval    = 60000000000
+}
+
+data "flux_sync" "groups" {
+  for_each = {
+    for ns in var.namespaces :
+    ns.name => ns
+    if ns.flux.enabled
+  }
+
+  url         = "http://git-cache-http-server/dev.azure.com/${var.azdo_org}/${var.azdo_proj}/_git/${each.key}"
+  branch      = var.branch
+  target_path = var.git_path
+  name        = each.key
+  interval    = 60000000000
+}
+
+# Azure DevOps
+data "azuredevops_project" "this" {
+  name = var.azdo_proj
+}
+
+resource "azuredevops_git_repository" "this" {
+  project_id = data.azuredevops_project.this.id
+  name       = var.azdo_repo
+  initialization {
+    init_type = "Clean"
+  }
+}
+
+resource "azuredevops_git_repository" "groups" {
+  for_each = {
+    for ns in var.namespaces :
+    ns.name => ns
+    if ns.flux.enabled
+  }
+
+  project_id = data.azuredevops_project.this.id
+  name       = each.key
+  initialization {
+    init_type = "Clean"
+  }
+}
+
+resource "azuredevops_git_repository_file" "install" {
+  repository_id = azuredevops_git_repository.this.id
+  file          = data.flux_install.main.path
+  content       = data.flux_install.main.content
+  branch        = "refs/heads/${var.branch}"
+}
+
+resource "azuredevops_git_repository_file" "sync" {
+  repository_id = azuredevops_git_repository.this.id
+  file          = data.flux_sync.main.path
+  content       = data.flux_sync.main.content
+  branch        = "refs/heads/${var.branch}"
+}
+
+resource "azuredevops_git_repository_file" "kustomize" {
+  repository_id = azuredevops_git_repository.this.id
+  file          = data.flux_sync.main.kustomize_path
+  content       = data.flux_sync.main.kustomize_content
+  branch        = "refs/heads/${var.branch}"
+}
+
+resource "azuredevops_git_repository_file" "groups" {
+  for_each = {
+    for ns in var.namespaces :
+    ns.name => ns
+    if ns.flux.enabled
+  }
+
+  repository_id = azuredevops_git_repository.this.id
+  file          = "${var.git_path}/${each.key}.yaml"
+  content       = data.flux_sync.groups[each.key].content
+  branch        = "refs/heads/${var.branch}"
+}
```

Comment:
@phillebaba what is this meant for?

Reply:
This is the sync configuration for each individual repository